### ISSUE TITLE
Modifying preset stage-0 fails

### ISSUE THREAD

**Issue by sorgloomer** [developit/modify-babel-preset#6]
Hi, I was trying to modify the stage-0 preset like

```
const babel_preset = require("modify-babel-preset")("stage-0", {
  "transform-es2015-modules-commonjs": false,
  "external-helpers": true
});
```

it failed with

```
resolve failed:  { Error: Cannot find module 'README.md'
  at Function.Module._resolveFilename (module.js:440:15)
  at Function.requireRelative.resolve (...\project\node_modules\require-relative\index.js:30:17)
  at resolve (...\project\node_modules\modify-babel-preset\lib\serialize.js:25:26)
  at findAndRemove (...\project\node_modules\modify-babel-preset\lib\serialize.js:67:11)
  at ...\project\node_modules\modify-babel-preset\lib\serialize.js:92:14
  at Array.map (native)
  at loadPreset (...\project\node_modules\modify-babel-preset\lib\serialize.js:90:29)
  at module.exports (...\project\node_modules\modify-babel-preset\index.js:88:19)
  at Object.module.exports.grunt (...\project\gruntfile.js:5:54)
  at loadTask (...\project\node_modules\grunt\lib\grunt\task.js:318:10)
  at Task.task.init (...\project\node_modules\grunt\lib\grunt\task.js:437:5)
  at Object.grunt.tasks (...\project\node_modules\grunt\lib\grunt.js:111:8)
  at Object.module.exports [as cli] (...\project\node_modules\grunt\lib\grunt\cli.js:27:9)
  at Object.<anonymous> (C:\Users\Hege\AppData\Roaming\npm\node_modules\grunt-cli\bin\grunt:44:20)
  at Module._compile (module.js:541:32)
  at Object.Module._extensions..js (module.js:550:10)
  at Module.load (module.js:458:32)
  at tryModuleLoad (module.js:417:12)
  at Function.Module._load (module.js:409:3)
  at Function.Module.runMain (module.js:575:10)
  at startup (node.js:160:18)
  at node.js:456:3
 code: 'MODULE_NOT_FOUND' } README.md
resolve failed:  { Error: Cannot find module 'package.json'
  at Function.Module._resolveFilename (module.js:440:15)
  at Function.requireRelative.resolve (...\project\node_modules\require-relative\index.js:30:17)
  at resolve (...\project\node_modules\modify-babel-preset\lib\serialize.js:25:26)
  at findAndRemove (...\project\node_modules\modify-babel-preset\lib\serialize.js:67:11)
  at ...\project\node_modules\modify-babel-preset\lib\serialize.js:92:14
  at Array.map (native)
  at loadPreset (...\project\node_modules\modify-babel-preset\lib\serialize.js:90:29)
  at module.exports (...\project\node_modules\modify-babel-preset\index.js:88:19)
  at Object.module.exports.grunt (...\project\gruntfile.js:5:54)
  at loadTask (...\project\node_modules\grunt\lib\grunt\task.js:318:10)
  at Task.task.init (...\project\node_modules\grunt\lib\grunt\task.js:437:5)
  at Object.grunt.tasks (...\project\node_modules\grunt\lib\grunt.js:111:8)
  at Object.module.exports [as cli] (...\project\node_modules\grunt\lib\grunt\cli.js:27:9)
  at Object.<anonymous> (C:\Users\Hege\AppData\Roaming\npm\node_modules\grunt-cli\bin\grunt:44:20)
  at Module._compile (module.js:541:32)
  at Object.Module._extensions..js (module.js:550:10)
  at Module.load (module.js:458:32)
  at tryModuleLoad (module.js:417:12)
  at Function.Module._load (module.js:409:3)
  at Function.Module.runMain (module.js:575:10)
  at startup (node.js:160:18)
  at node.js:456:3
 code: 'MODULE_NOT_FOUND' } package.json
resolve failed:  { Error: Cannot find module 'index.js'
  at Function.Module._resolveFilename (module.js:440:15)
  at Function.requireRelative.resolve (...\project\node_modules\require-relative\index.js:30:17)
  at resolve (...\project\node_modules\modify-babel-preset\lib\serialize.js:25:26)
  at findAndRemove (...\project\node_modules\modify-babel-preset\lib\serialize.js:67:11)
  at ...\project\node_modules\modify-babel-preset\lib\serialize.js:92:14
  at Array.map (native)
  at loadPreset (...\project\node_modules\modify-babel-preset\lib\serialize.js:90:29)
  at module.exports (...\project\node_modules\modify-babel-preset\index.js:88:19)
  at Object.module.exports.grunt (...\project\gruntfile.js:5:54)
  at loadTask (...\project\node_modules\grunt\lib\grunt\task.js:318:10)
  at Task.task.init (...\project\node_modules\grunt\lib\grunt\task.js:437:5)
  at Object.grunt.tasks (...\project\node_modules\grunt\lib\grunt.js:111:8)
  at Object.module.exports [as cli] (...\project\node_modules\grunt\lib\grunt\cli.js:27:9)
  at Object.<anonymous> (C:\Users\Hege\AppData\Roaming\npm\node_modules\grunt-cli\bin\grunt:44:20)
  at Module._compile (module.js:541:32)
  at Object.Module._extensions..js (module.js:550:10)
  at Module.load (module.js:458:32)
  at tryModuleLoad (module.js:417:12)
  at Function.Module._load (module.js:409:3)
  at Function.Module.runMain (module.js:575:10)
  at startup (node.js:160:18)
  at node.js:456:3
 code: 'MODULE_NOT_FOUND' } index.js
resolve failed:  TypeError: Cannot read property 'length' of undefined
  at Function.Module._resolveLookupPaths (module.js:290:23)
  at Function.Module._resolveFilename (module.js:431:31)
  at Function.requireRelative.resolve (...\project\node_modules\require-relative\index.js:30:17)
  at resolve (...\project\node_modules\modify-babel-preset\lib\serialize.js:25:26)
  at loadPreset (...\project\node_modules\modify-babel-preset\lib\serialize.js:81:13)
  at ...\project\node_modules\modify-babel-preset\lib\serialize.js:94:11
  at Array.map (native)
  at loadPreset (...\project\node_modules\modify-babel-preset\lib\serialize.js:90:29)
  at module.exports (...\project\node_modules\modify-babel-preset\index.js:88:19)
  at Object.module.exports.grunt (...\project\gruntfile.js:5:54)
  at loadTask (...\project\node_modules\grunt\lib\grunt\task.js:318:10)
  at Task.task.init (...\project\node_modules\grunt\lib\grunt\task.js:437:5)
  at Object.grunt.tasks (...\project\node_modules\grunt\lib\grunt.js:111:8)
  at Object.module.exports [as cli] (...\project\node_modules\grunt\lib\grunt\cli.js:27:9)
  at Object.<anonymous> (C:\Users\Hege\AppData\Roaming\npm\node_modules\grunt-cli\bin\grunt:44:20)
  at Module._compile (module.js:541:32)
  at Object.Module._extensions..js (module.js:550:10)
  at Module.load (module.js:458:32)
  at tryModuleLoad (module.js:417:12)
  at Function.Module._load (module.js:409:3)
  at Function.Module.runMain (module.js:575:10)
  at startup (node.js:160:18)
  at node.js:456:3
 undefined
```

Is anyone else experiencing this?

**Comment by developit** [comment]
@sorgloomer I'm wondering if this is a Windows issue.  Are you able to post the full module somewhere?  (or at least the `package.json`)

**Comment by sorgloomer** [comment]
Hmm it neither works with es2015 for me. I must miss some trivial thing.

`package.json`

```
{
  "devDependencies": {
    "babel-plugin-external-helpers": "^6.8.0",
    "babel-preset-es2015": "^6.9.0",
    "modify-babel-preset": "^2.0.3"
  }
} 
```

`index.js`

```
require("modify-babel-preset")("es2015", {
  "transform-es2015-modules-commonjs": false,
  "external-helpers": true
});
```

Stack trace is very long, the top of it is the same:

```
resolve failed:  { Error: Cannot find module 'README.md'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.requireRelative.resolve (C:\_temp\asdasd\node_modules\require-relative\index.js:30:17)
    at resolve (C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:25:26)
    at findAndRemove (C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:67:11)
    at C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:108:13
    at Array.map (native)
    at loadPreset (C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:100:29)
    at module.exports (C:\_temp\asdasd\node_modules\modify-babel-preset\index.js:88:19)
    at Object.<anonymous> (C:\_temp\asdasd\index.js:1:93)
    at Module._compile (module.js:541:32) code: 'MODULE_NOT_FOUND' } README.md
resolve failed:  { Error: Cannot find module 'package.json'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.requireRelative.resolve (C:\_temp\asdasd\node_modules\require-relative\index.js:30:17)
    at resolve (C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:25:26)
    at findAndRemove (C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:67:11)
    at C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:108:13
    at Array.map (native)
    at loadPreset (C:\_temp\asdasd\node_modules\modify-babel-preset\lib\serialize.js:100:29)
    at module.exports (C:\_temp\asdasd\node_modules\modify-babel-preset\index.js:88:19)
    at Object.<anonymous> (C:\_temp\asdasd\index.js:1:93)
    at Module._compile (module.js:541:32) code: 'MODULE_NOT_FOUND' } package.json
...
```

Running node v6.2.1 on 64-bit windows 10

**Comment by developit** [comment]
@sorgloomer A couple things:  your `package.json` needs to specify `index.js` as the `"main"` field.

Also, you need to export the configuration:

``` js
module.exports = require("modify-babel-preset")("es2015", {
  "transform-es2015-modules-commonjs": false,
  "external-helpers": true
});
```

Lastly, and I imagine this is causing the issue above - all three dependencies must be listed as `"dependencies"` (or optionally `"peerDependencies"`).  Installing them as `"devDependencies"` means they are not installed when the module is used by another module.

Here's a gist showing what worked for me:
https://gist.github.com/developit/a224864f3629e743e9ad04b6175633b3

**Comment by sorgloomer** [comment]
I didn't bother myself with creating a correct package.json as I thought it was irrelevant. And for me neither works the gist you've sent. I do not import the module, but run index.js directly. I don't think it should make a difference right?

So here is what I did: created a new empty folder, created package.json and index.js from your gist, ran npm install and node index.js, and the same very very long stacktrace popped out in the console.

Is it possible it is an issue with windows?

**Comment by sorgloomer** [comment]
Same experiment works on ubuntu without an error. On windows, requiring the module as it was an installed package produces the same error.

**Comment by developit** [comment]
Definitely seems like a Windows issue. I'll check if there are any hard-coded path separators.

**Comment by sorgloomer** [comment]
It looks this only occurs when modify-babel-preset is used directly from the entry package.

**Comment by developit** [comment]
This should be fixed as of `modify-babel-preset@3.0.0`.  Let me know if it's still broken!

**Comment by sorgloomer** [comment]
Looks good, thank you so much!
